api/provider/aws/deploy_list.go

```diff
@@ -1,9 +1,69 @@
 package aws
 
 import (
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ecs"
 	"github.com/quintilesims/layer0/common/models"
 )
 
 func (d *DeployProvider) List() ([]models.DeploySummary, error) {
-	return nil, nil
+	taskDefinitionARNs, err := d.listTaskDefinitionARNs()
+	if err != nil {
+		return nil, err
+	}
+
+	summaries, err := d.populateSummariesFromTaskDefinitionARNs(taskDefinitionARNs)
+	if err != nil {
+		return nil, err
+	}
+
+	return summaries, nil
+}
+
+func (d *DeployProvider) listTaskDefinitionARNs() ([]string, error) {
+	taskDefinitionARNs := []string{}
+	fn := func(output *ecs.ListTaskDefinitionsOutput, lastPage bool) bool {
+		for _, taskDefinitionARN := range output.TaskDefinitionArns {
+			taskDefinitionARNs = append(taskDefinitionARNs, aws.StringValue(taskDefinitionARN))
+		}
+		return !lastPage
+	}
+
+	input := &ecs.ListTaskDefinitionsInput{}
+	if err := d.AWS.ECS.ListTaskDefinitionsPages(input, fn); err != nil {
+		return nil, err
+	}
+
+	return taskDefinitionARNs, nil
+}
+
+func (d *DeployProvider) populateSummariesFromTaskDefinitionARNs(taskDefinitionARNs []string) ([]models.DeploySummary, error) {
+	deployTags, err := d.TagStore.SelectByType("deploy")
+	if err != nil {
+		return nil, err
+	}
+
+	summaries := make([]models.DeploySummary, 0, len(taskDefinitionARNs))
+	for _, taskDefinitionARN := range taskDefinitionARNs {
+		for _, tag := range deployTags.WithKey("arn") {
```

Comment:
You should use a map to check for matches, see: https://github.com/quintilesims/layer0/blob/968608cc69771321bb1b219b9694a2e9c8b8bb07/api/provider/aws/task_list.go#L64

Reply:
:white_check_mark: